pkg/utils/disk_image/guestfs.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2016 Mirantis
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package disk_utils
+
+
+import (
+	"libguestfs.org/guestfs"
```

Comment:
Yep, I would be glade if it would be possible. :(
The thing is that's tricky: there is c-library guestfs, the project on github is https://github.com/libguestfs/libguestfs, and there is a generator writtin in OCaml within this project (https://github.com/libguestfs/libguestfs/tree/master/generator) which generate bindings for the set of the language including python and go, i.e. you need to build it all to have it.
The destribution of these bindings is happening within packages. For each language - separate package. For go it consists of only one go module with LGPL lisence which is placed in /usr/share/gocode, but it has a lot of useful wrappers, especially for structures, what lets to use guestfs more easily then thru cgo.

So as I see the best option here is just to take go file from official package and just to attach it to our project with checking depency on libguestfs0 lib version.

Other options I see:
- use cgo
- work with packages

WDYT?